### PR TITLE
AAG-2777: Added SwiftUI example app

### PR DESCRIPTION
**Pull request by me2k**
## JIRA
[AAG-2777]

## DESCRIPTION
Added SwiftUI example app as a new target to the project. This lives independently to the existing test app.

## SCREENSHOTS
Simulator Screen Shot - iPhone 14 Pro - 2023-01-25 at 16 20 34
Simulator Screen Shot - iPhone 14 Pro - 2023-01-25 at 16 20 38
Simulator Screen Shot - iPhone 14 Pro - 2023-01-25 at 16 20 44
Simulator Screen Shot - iPhone 14 Pro - 2023-01-25 at 16 20 45
Simulator Screen Shot - iPhone 14 Pro - 2023-01-25 at 16 20 48
Simulator Screen Shot - iPhone 14 Pro - 2023-01-25 at 16 20 54

[AAG-2777]: https://superawesomeltd.atlassian.net/browse/AAG-2777?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Example/SuperAwesomeSwiftUIExample/Enums/BannerType.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//
+//  BannerType.swift
+//  AwesomeAdsSDKKitchenSink
+//
+//  Created by Myles Eynon on 18/01/2023.
+//
+
+import Foundation
```
Comment (thomas-james-1986) [inline Example/SuperAwesomeSwiftUIExample/Enums/BannerType.swift:8]: Is this file needed?

File: Example/SuperAwesomeSwiftUIExample/Enums/FeatureType.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    case bumper
+    case interstial
+    case parentGate
+    case video
+
+    var title: String {
+        var type = ""
+        switch self {
+        case .interstial:
+            type = "Interstital"
```
Comment (thomas-james-1986) [inline Example/SuperAwesomeSwiftUIExample/Enums/FeatureType.swift:21]: I think you could do `rawValue.capitalized` here?

File: Example/SuperAwesomeSwiftUIExample/Enums/BannerType.swift (hunk 1 above)
Comment (me2k) [reply Example/SuperAwesomeSwiftUIExample/Enums/BannerType.swift]: It's gone, I was experimenting in my example and forgot to remove it.

File: Example/SuperAwesomeSwiftUIExample/Enums/FeatureType.swift (hunk 2 above)
Comment (me2k) [reply Example/SuperAwesomeSwiftUIExample/Enums/FeatureType.swift:19]: We can improve it for sure, I'll do just that 👍

Comment (thomas-james-1986) [reply Example/SuperAwesomeSwiftUIExample/Enums/FeatureType.swift:19]: You could use `String` as the base type, it would be cleaner actually

File: Example/SuperAwesomeSwiftUIExample/Enums/FeatureType.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        case .video:
+            type = "Video"
+        default:
+            type = "Banner"
+        }
+        return type
+    }
+}
```
Comment (thomas-james-1986) [inline Example/SuperAwesomeSwiftUIExample/Enums/FeatureType.swift:27]: Would this work as you intend?

```
enum FeatureType: String {
     case banner
     case interstial
     case video

     var title: String {
         return rawValue.capitalized
     }
 }
```

**Review by thomas-james-1986 — COMMENTED** [review]
Looks great! 👏🚀 just wondering if there's milage in the capitalisation suggestion 😂

**Review by thomas-james-1986 — APPROVED** [review]
🙌
